### PR TITLE
fix EthLatestBlockNumber value

### PR DESCRIPTION
**Pull request by thomas-nguy**
Closes: #297

## Description

Currently we are always missing the logs in the latest block, this is cause by the value of EthLatestBlockNumber which return latest block -1

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: ethereum/rpc/backend/backend.go [hunk 1]
```diff
@@ -241,7 +241,7 @@ func (e *EVMBackend) HeaderByNumber(blockNum types.BlockNumber) (*ethtypes.Heade
 	switch blockNum {
 	case types.EthLatestBlockNumber:
 		if currentBlockNumber > 0 {
-			height = int64(currentBlockNumber - 1)
+			height = int64(currentBlockNumber)
 		}
 	case types.EthPendingBlockNumber:
 		if currentBlockNumber > 0 {
```
Comment (thomas-nguy) [inline ethereum/rpc/backend/backend.go:247]: for `pendingBlockNumber`, not sure which value we should return

Comment (crypto-facs) [reply ethereum/rpc/backend/backend.go:247]: I think there might have to be a global solution for this.. as there are different endpoints that face a similar issue due to the same problem [more info here](https://github.com/tharsis/ethermint/issues/215)

Comment (yihuang) [reply ethereum/rpc/backend/backend.go:247]: I think we don't have the concept of pending block here, so it should just be the same as latest one.

**Review by leejw51crypto — APPROVED** [review]
lgtm

**Review by yihuang — APPROVED** [review]

File: ethereum/rpc/backend/backend.go (hunk 1 above)
Comment (fedekunze) [reply ethereum/rpc/backend/backend.go:247]: yeah there is no pending block, just "pending" txs from the mempool

**Review by fedekunze — APPROVED** [review]

**Pull request merged**